### DISCUSSION TITLE
Suggestion: download album as ZIP

### DISCUSSION THREAD

**Discussion by absolutelynothinghere** [Chevereto/Chevereto-Free#170] [Ideas]
Hello! Is it possible to let users one-click download an entire album as a ZIP file? That would be a nice feature.

**Comment by rodber** [comment]
No, is not possible at this time as I believe it could be easily exploited for ddos.

Storage in Chevereto is distributed among many **different** external servers (paid edition) so to openly offer album downloads gets expensive for users as it requires to fetch and download these every-time the zip file needs to be created/updated. An alternative could be to drive this on the web browser layer (as fetch + zip the files on the web browser) but that is also expensive as web browsers can't allocate your entire system resources, it is limited making the experience more unusable as the bigger the album gets.

The only feasible solution for our realm it is to use the API to orchestrate the production of these zip files by running everything on the client device, namely image fetchs and zipping.

There are a few third-party software for this purpose, heres one: https://chevereto.com/community/threads/chevereto-album-downloader.13711/

There's also browser extensions, these will work fine long as the album isn't huge.

- https://chrome.google.com/webstore/detail/chevereto-download-button/lgcgmhfenjjeghjipookhcplpifakgcd?hl=en
- https://addons.mozilla.org/en-US/firefox/addon/chevereto-download-button/
